options.go

```diff
@@ -2,40 +2,27 @@ package discovery
 
 import (
 	"time"
-)
 
-// Option is a single discovery option
-type Option func(opts *Options) error
+	moved "github.com/libp2p/go-libp2p-core/discovery"
```

Comment:
I would call this something else in here, `moved` is ugly and we use these types.
Perhaps `iface`?

Reply:
`iface` won't work as the target packages doesn't only contain interfaces. I think `moved` is fine as it clearly conveys intent. Alternatively `target`, but that's less specific.

Reply:
Just call it `core`!

Reply:
Thinking of it, `core` is not half-bad, as it conveys where the type lives now as well.

Reply:
Ok, you won, I've changed it everywhere to `core`.